js/ui/components/clubhouse.js

```diff
@@ -874,7 +877,16 @@ var ClubhouseComponent = new Lang.Class({
     },
 
     callShow: function(timestamp) {
-        this.proxy.showRemote(timestamp);
+        if (this.proxy.g_name_owner) {
+            this.proxy.showRemote(timestamp);
+            return;
+        }
+
+        // We only activate the app here if it's not yet running, otherwise the cursor will turn
+        // into a spinner for a while, even after the window is shown.
+        // @todo: Use the activate logic alone when we fix the problem mentioned above.
+        let activationContext = new AppActivation.AppActivationContext(this._clubhouseApp);
+        activationContext.activate();
```

Comment:
You should pass the timestamp here, it will be used by `activate()`.
However we don't need any of the complexity of `AppActivationContext` which exists to deal with the splash screens, and using it could potentially introduce issues down the line...

I'd prefer it if you called directly `activate()` or `activate_full()` on the `ShellApp`.